### DISCUSSION TITLE
设置color数据驱动数据量一两百个时，chart.render渲染特别慢

### DISCUSSION THREAD

**Discussion by ChristmasFox** [antvis/G2#5213] [Q&A]
把图表库引到自己的系统后，color通道字段离散值一两百个，render执行特别慢，但是同样的options放到G2官网上render特别快，速度能差十倍，（另外数据量大的时候，不设置color的话，也会比官网慢一些，不知为何），请问有没有解决方案
![3661687251563_ pic](https://github.com/antvis/G2/assets/68337739/7d6ff66a-6b0c-4bee-8d27-da2bf8355c70)
控制台出现警告
![3701687251654_ pic](https://github.com/antvis/G2/assets/68337739/9d0a9f62-8966-4622-a036-d1545c2a2551)

**Comment by pearmini** [comment]
是开了图例吗？

**Reply by ChristmasFox** [reply]
对开了图例

**Reply by ChristmasFox** [reply]
官网上使用的也开了图例 很快，刚才试了下在我们系统中关了图例确实快了很多

**Reply by pearmini** [reply]
大概有多少个图例？

**Reply by pearmini** [reply]
可以提供一个 codesanbox 的 demo 吗？

**Reply by ChristmasFox** [reply]
二百个

**Reply by ChristmasFox** [reply]
demo稍等我创建一个

**Reply by ChristmasFox** [reply]
[示例](https://codesandbox.io/s/vigilant-cohen-snx68t?file=/src/index.js)

**Reply by pearmini** [reply]
好的
